### DISCUSSION TITLE
[Help]: I am making cards carousel, each card has shadow, but the shadow is cutoff due to carousel wrapper any workaround?

### DISCUSSION THREAD

**Discussion by abdelrahman-abdelmoaty** [davidjerleke/embla-carousel#1016] [Help] [answered]
### Summary

[Help]: I am making cards carousel, each card has shadow, but the shadow is cutoff due to carousel wrapper any work around?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by abdelrahman-abdelmoaty** [comment]
I solved the vertical shadow cut off by using padding to the carousel wrapper, tried the same for horizonal but keep giving me bugs in alighnment with autoplay.

**Comment by abdelrahman-abdelmoaty** [comment]
![image](https://github.com/user-attachments/assets/d89c1b68-fa79-429e-9ee9-a98a762d0a2d)

**Comment by abdelrahman-abdelmoaty** [comment]
When applying px and -mx it works only for first slide and then slides go out of container
![Screenshot_20](https://github.com/user-attachments/assets/8175dccf-5f4d-4186-a302-31da67e2eddc)
![image](https://github.com/user-attachments/assets/caec4b1a-271b-4176-9597-13f32b104326)

**Reply by davidjerleke** [reply, marked as answer]
Hi @Abdelrahman-Abdelmoaty,

Building on one of the examples on the documentation website, you can solve it like [this](https://codesandbox.io/p/sandbox/embla-carousel-loop-react-forked-v7sc96?file=%2Fsrc%2Fcss%2Fembla.css%3A32%2C1-34%2C54).

**Reply by abdelrahman-abdelmoaty** [reply]
@davidjerleke thanks I did something close to that, have another issue, I am using 4 carousels with autoplay in my home page nextjs app and it feels like lagging, especially in safari, do you have any article, discussion how to handle these performance issues.

**Reply by davidjerleke** [reply]
@Abdelrahman-Abdelmoaty this is a different question, not relevant here, but only enable Autoplay on carousels that are in the viewport.

**Reply by abdelrahman-abdelmoaty** [reply]
Ok mb, is there a built in prop for start play it if it is in viewport?

**Reply by davidjerleke** [reply]
@Abdelrahman-Abdelmoaty no, join or follow this discussion instead of bloating this with irrelevant comments:

- #1010

Thanks for understanding.
